src/wp-includes/ms-functions.php

```diff
@@ -644,8 +587,8 @@ function wpmu_validate_blog_signup( $blogname, $blog_title, $user = '' ) {
 		$errors->add( 'blogname', __( 'Please enter a site name.' ) );
 	}
 
-	if ( preg_match( '/[^a-z0-9]+/', $blogname ) ) {
-		$errors->add( 'blogname', __( 'Site names can only contain lowercase letters (a-z) and numbers.' ) );
+	if ( preg_match( '/[^a-z0-9-]+/', $blogname ) ) {
+		$errors->add( 'blogname', __( 'Site names can only contain lowercase letters (a-z), numbers, and dashes.' ) );
```

Comment:
This will need to be modified slightly.

The DNS system doesn't alllow for domains/subdomains to start or end with a hyphen.

```suggestion
	if ( preg_match( '/[^a-z0-9-]+/', $blogname ) ) {
		$errors->add( 'blogname', __( 'Site names can only contain lowercase letters (a-z), numbers, and hyphens.' ) );
	} elseif ( trim( $blogname, '-' ) !== $blogname  ) {
		$errors->add( 'blogname', __( 'Site names can not begin or end with a hyphen.' ) );
```